### PR TITLE
Inagurator upgrade to kernel 4.11 and NVDIMM support

### PR DESCRIPTION
**Pull request by ron-lb**
Upgraded Inaugurator kernel to 4.11.Based on Fedora 25.

Original issue was: IRS-237/LBM1-1586 NVDIMM devices were not recognized after inauguration.
This required a new kexec version (see https://github.com/horms/kexec-tools/commit/cb9a818ff2ac437e76512ec01c0eb22d0d7456b2). As part of resolving additional issue, also upgraded the kernel version.

### CODE REVIEW

File: Makefile.build [hunk 1]
```diff
@@ -67,7 +67,7 @@ build/inaugurator.%.initrd.img: $(INITRD_IMAGE_PATH)
 	mv $@.tmp $@
 
 build/osmosis-1.0.linux-x86_64.tar.gz:
-	(cd ../osmosis/ && $(MAKE) clean && $(MAKE) build && $(MAKE) egg)
+	(cd ../osmosis/ && $(MAKE) clean && $(MAKE) -j3 build && $(MAKE) egg)
```
Comment (sashas-lb) [inline Makefile.build]: please remove it from PR .. this will be passed from outer makefile automatically and when built on laptop it slows it down too much

File: Makefile.build [hunk 2]
```diff
@@ -120,22 +120,20 @@ build/inaugurator.thin.initrd.dir: ${PYTHON_DIST_LOCAL_PATH} build/osmosis-1.0.l
...
 	DEST=$@.tmp KERNEL_UNAME_R=$(KERNEL_VERSION) sh/relative_copy_driver.sh ixgbe
 	DEST=$@.tmp KERNEL_UNAME_R=$(KERNEL_VERSION) sh/relative_copy_driver.sh virtio_blk
-	DEST=$@.tmp KERNEL_UNAME_R=$(KERNEL_VERSION) sh/relative_copy_driver.sh isci
 	DEST=$@.tmp KERNEL_UNAME_R=$(KERNEL_VERSION) sh/relative_copy_driver.sh virtio_net
 	DEST=$@.tmp KERNEL_UNAME_R=$(KERNEL_VERSION) sh/relative_copy_driver.sh virtio_pci
 	DEST=$@.tmp KERNEL_UNAME_R=$(KERNEL_VERSION) sh/relative_copy_driver.sh e1000
 	DEST=$@.tmp KERNEL_UNAME_R=$(KERNEL_VERSION) sh/relative_copy_driver.sh igb
 	DEST=$@.tmp KERNEL_UNAME_R=$(KERNEL_VERSION) sh/relative_copy_driver.sh 8139cp
 	DEST=$@.tmp KERNEL_UNAME_R=$(KERNEL_VERSION) sh/relative_copy_driver.sh vfat
-	DEST=$@.tmp KERNEL_UNAME_R=$(KERNEL_VERSION) sh/relative_copy_driver.sh ext4
```
Comment (sashas-lb) [inline Makefile.build]: why are you removing drivers ? 
if not needed to remove why removing ? if needed lets split it into different commit explaining why

File: docker/build-inaugurator.dockerfile [hunk 3]
```diff
@@ -1,31 +1,21 @@
...
-
-RUN yum update -y
-
-# Add the EPEL repository and update all packages
-RUN curl http://dl.fedoraproject.org/pub/epel/7/x86_64/e/epel-release-7-9.noarch.rpm -o temp && \
-    rpm -ivh temp && \
-    rm temp
-
 # Install other tools
-RUN yum update -y
```
Comment (sashas-lb) [inline docker/build-inaugurator.dockerfile]: is this change needed ? if not really needed i prefer to stay close to stratoscale version so that we can easily rebase their fixes..

File: docker/build-inaugurator.dockerfile [hunk 4]
```diff
@@ -1,31 +1,21 @@
...
+RUN dnf install -y \
     sudo \
     boost-devel \
     boost-static \
     openssl-devel \
     gcc-c++ \
     hwdata \
-    kexec-tools \
     net-tools \
+    ethtool \
```
Comment (sashas-lb) [inline docker/build-inaugurator.dockerfile]: why did you add ethtool ? if not needed lets remove if needed lets move to different commit explaining why

File: inaugurator/partitiontable.py [hunk 5]
```diff
@@ -14,12 +14,12 @@ class PartitionTable:
         minimumRoot=7,
         createRoot=10)
     VOLUME_GROUP = "inaugurator"
-    LAYOUT_SCHEMES = dict(GPT=dict(partitions=dict(bios_boot=dict(sizeMB=2, flags="bios_grub"),
```
Comment (sashas-lb) [inline inaugurator/partitiontable.py]: why this file had been changed ? please revert this

File: inaugurator/udev.py [hunk 6]
```diff
@@ -28,7 +28,7 @@ def loadAllDrivers():
 def _loadDriver(driver):
     "This is for upwards dependency, not modprobe like dependency"
     print "Driver: %s, modprobing" % driver
-    sh.run("busybox modprobe %s" % driver)
+    sh.run("modprobe %s" % driver)
```
Comment (sashas-lb) [inline inaugurator/udev.py]: why this change is needed ?

File: sh/list_storage_and_network_driver_names.sh [hunk 7]
```diff
@@ -3,4 +3,4 @@
 BLACKLIST='-e bfa -e csiostor -e cxgb4 -e bna -e aic94xx'
 KERNEL_VERSION=$1
 set -e
-find /lib/modules/$KERNEL_VERSION/kernel/drivers/net/ethernet /lib/modules/$KERNEL_VERSION/kernel/drivers/scsi -type f -printf '%f\n' | sed 's/\.ko$//' | grep -v $BLACKLIST
+find /lib/modules/$KERNEL_VERSION/kernel/drivers/net/ethernet /lib/modules/$KERNEL_VERSION/kernel/drivers/scsi -type f -printf '%f\n' | sed 's/\.ko.xz$//' | grep -v $BLACKLIST
```
Comment (sashas-lb) [inline sh/list_storage_and_network_driver_names.sh]: what is this change for ?

File: sh/relative_copy_driver.sh [hunk 8]
```diff
@@ -13,5 +13,7 @@ for ko in `modprobe --show-depends $1 --set-version=$KERNEL_UNAME_R | sed 's/ins
 done
 modinfo --field firmware $1 --set-version=$KERNEL_UNAME_R >& /dev/null
 for firmware in `modinfo --field firmware $1 --set-version=$KERNEL_UNAME_R | grep -v $FIRMWARE_BLACKLIST`; do
+if [ -e /lib/firmware/$firmware ]; then
```
Comment (sashas-lb) [inline sh/relative_copy_driver.sh]: why is this done ? can you please explain and move to separate commit ?

**Comment by sashas-lb** [comment]
@ron-lb  all in all great changes, please work little bit on commit split and removing uneeded changes .. we prefer to do as minimum changes as needed here to stay relatively close to stratoscale

File: docker/build-inaugurator.dockerfile (hunk 3 above)
Comment (maor-lb) [inline docker/build-inaugurator.dockerfile]: @sashas-lb  dnf replaced yum so it's needed

File: inaugurator/udev.py (hunk 6 above)
Comment (maor-lb) [inline inaugurator/udev.py]: @sashas-lb  please read the commit message it explain why (really some kind of mystery)

File: sh/list_storage_and_network_driver_names.sh (hunk 7 above)
Comment (maor-lb) [inline sh/list_storage_and_network_driver_names.sh]: fedora kernel using  xz ko

File: sh/relative_copy_driver.sh (hunk 8 above)
Comment (maor-lb) [inline sh/relative_copy_driver.sh]: +1 for @sashas-lb comment

File: inaugurator/udev.py (hunk 6 above)
Comment (sashas-lb) [inline inaugurator/udev.py]: great i think this deserves commit of itself .. @ron-lb can you please split it to different commit ..

**Comment by sashas-lb** [comment]
@maor-lb  we dont  really dont need lab maitenance window to deploy it ... we only need to replace boot docker .. it wont disturb the lab

File: Makefile.build (hunk 2 above)
Comment (ron-lb) [inline Makefile.build]: These are builtin in the fedora image. Will split to different commit.

File: inaugurator/partitiontable.py (hunk 5 above)
Comment (ron-lb) [inline inaugurator/partitiontable.py]: The problem is that the boot partition is created with a boot flag, but when queried it reports "boot, eps" which causes the check to fail. As a quick workaround I separated between the flags used at creation and flags used at verification.

**Comment by sashas-lb** [comment]
Can you please split it to different commit

File: Makefile.build (hunk 2 above)
Comment (ron-lb) [inline Makefile.build]: Done

File: inaugurator/partitiontable.py (hunk 5 above)
Comment (ron-lb) [inline inaugurator/partitiontable.py]: Moved to separate commit as shasha requested

File: docker/build-inaugurator.dockerfile [hunk 9]
```diff
@@ -34,19 +23,15 @@ RUN yum install -y \
     make \
     kernel \
     lshw \
-    rsync && \
-    yum -y clean all
+    rsync \
+    busybox 2>&1 && \
```
Comment (sashas-lb) [inline docker/build-inaugurator.dockerfile]: what is this ? why busybox 2>&1????

File: setup.py [hunk 10]
```diff
@@ -7,7 +7,7 @@ def read(fname):
 
 setup(
     name="inaugurator",
-    version="1.3",
+    version="1.4",
```
Comment (sashas-lb) [inline setup.py]: please bump revision as last commit .. :)

File: Makefile.build [hunk 11]
```diff
@@ -120,22 +120,20 @@ build/inaugurator.thin.initrd.dir: ${PYTHON_DIST_LOCAL_PATH} build/osmosis-1.0.l
...
 	DEST=$@.tmp KERNEL_UNAME_R=$(KERNEL_VERSION) sh/relative_copy_driver.sh usb_storage
-	DEST=$@.tmp KERNEL_UNAME_R=$(KERNEL_VERSION) sh/relative_copy_driver.sh dm_mod
-	DEST=$@.tmp KERNEL_UNAME_R=$(KERNEL_VERSION) sh/relative_copy_driver.sh sd_mod
-	DEST=$@.tmp KERNEL_UNAME_R=$(KERNEL_VERSION) sh/relative_copy_driver.sh ahci
 	DEST=$@.tmp KERNEL_UNAME_R=$(KERNEL_VERSION) sh/relative_copy_driver.sh isofs
-	DEST=$@.tmp KERNEL_UNAME_R=$(KERNEL_VERSION) sh/relative_copy_driver.sh ata_piix
+	DEST=$@.tmp KERNEL_UNAME_R=$(KERNEL_VERSION) sh/relative_copy_driver.sh dax
+	DEST=$@.tmp KERNEL_UNAME_R=$(KERNEL_VERSION) sh/relative_copy_driver.sh dax_pmem
+	DEST=$@.tmp KERNEL_UNAME_R=$(KERNEL_VERSION) sh/relative_copy_driver.sh nd_pmem
+	DEST=$@.tmp KERNEL_UNAME_R=$(KERNEL_VERSION) sh/relative_copy_driver.sh nd_blk
```
Comment (sashas-lb) [inline Makefile.build]: can you plsease split .. removal of modules that are built in and adding of pmem modules into different commints

File: Makefile.build (hunk 2 above)
Comment (ira-lb) [inline Makefile.build]: makes little sense to split this from the kernel version change, but whatevs.

Comment (ron-lb) [inline Makefile.build]: Done

File: setup.py (hunk 10 above)
Comment (ron-lb) [inline setup.py]: Done

File: docker/build-inaugurator.dockerfile (hunk 9 above)
Comment (ron-lb) [inline docker/build-inaugurator.dockerfile]: Fixed

File: Makefile.build (hunk 11 above)
Comment (ron-lb) [inline Makefile.build]: Done (no longer adding the pmem modules to the inaugurator image - they are not needed)
